jetty-openid/src/main/java/org/eclipse/jetty/security/openid/CredentialsDecoder.java

```diff
@@ -0,0 +1,91 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Map;
+
+import org.eclipse.jetty.util.ajax.JSON;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+/**
+ * Used to decode the ID Token from the base64 encrypted JSON Web Token (JWT).
+ */
+public class CredentialsDecoder
+{
+    private static final Logger LOG = Log.getLogger(CredentialsDecoder.class);
+    private static final Base64.Decoder decoder = Base64.getUrlDecoder();
```

Comment:
is this thread-safe? I think that `Base64.getUrlDecoder` already manages its state to make subsequent calls faster than the initial one. Not 100% about that though.

Reply:
I think it is thread-safe because `Base64.getUrlDecoder()` returns you its static field `RFC4648_URLSAFE`
where `static final Decoder RFC4648_URLSAFE = new Decoder(true, false);`

but I have changed it to a local variable which is probably clearer